### PR TITLE
core: process rcv file description

### PR DESCRIPTION
**Pull request by spaced4ndy**

### CODE REVIEW

File: src/Simplex/Chat/Store.hs [hunk 1]
```diff
@@ -2896,47 +2898,122 @@ deleteSndFileChunks :: DB.Connection -> SndFileTransfer -> IO ()
...
+
+getRcvFileDescrByFileId_ :: DB.Connection -> FileTransferId -> IO (Maybe RcvFileDescr)
+getRcvFileDescrByFileId_ db fileId =
+  maybeFirstRow toRcvFileDescr $
+    DB.query
+      db
+      [sql|
+        SELECT d.file_descr_id, d.file_descr_text, d.file_descr_part_no, d.file_descr_complete
+        FROM xftp_file_descriptions d
+        JOIN rcv_files f ON f.file_descr_id = d.file_descr_id
```
Comment (spaced4ndy) [inline src/Simplex/Chat/Store.hs:3003]: add indexes?

File: src/Simplex/Chat.hs [hunk 2]
```diff
@@ -2878,35 +2881,49 @@ processAgentMessageConn user@User {userId} corrId agentConnId agentMessage = do
...
+    groupMessageFileDescription GroupInfo {groupId} _m sharedMsgId fileDescr _msgMeta = do
+      fileId <- withStore $ \db -> getGroupFileIdBySharedMsgId db userId groupId sharedMsgId
+      processFDMessage fileId fileDescr
+
+    processFDMessage :: FileTransferId -> FileDescr -> m ()
+    processFDMessage fileId fileDescr = do
+      xftpCfg <- readTVarIO =<< asks userXFTPFileConfig
+      case xftpCfg of
+        Nothing -> pure ()
+        Just XFTPFileConfig {tempDirectory} -> do
```
Comment (epoberezkin) [inline src/Simplex/Chat.hs:2899]: probably temp folder needs to be split to a separate config, чтобы он всегда был (хотя бы как nothing).

File: src/Simplex/Chat.hs [hunk 3]
```diff
@@ -2878,35 +2881,49 @@ processAgentMessageConn user@User {userId} corrId agentConnId agentMessage = do
...
+          (rfd, _aci) <- withStore $ \db -> do
+            rfd <- appendRcvFD db userId fileId fileDescr
+            aci <- getChatItemByFileId db user fileId
+            -- ? re-create file item if it does not exist
```
Comment (epoberezkin) [inline src/Simplex/Chat.hs:2903]: ignore file if there is no chat item

File: src/Simplex/Chat/Store.hs [hunk 4]
```diff
@@ -2896,47 +2898,122 @@ deleteSndFileChunks :: DB.Connection -> SndFileTransfer -> IO ()
...
     (userId, contactId, fileName, fileSize, chunkSize, fileInline, CIFSRcvInvitation, currentTs, currentTs)
   fileId <- insertedRowId db
-  rfd <- mapM (createRcvFD_ db) fileDescr
-  let rfdId = (fileDescrId :: RcvFileDescr -> Int64) <$> rfd
-  DB.execute
-    db
-    "INSERT INTO rcv_files (file_id, file_status, file_queue_info, file_inline, rcv_file_inline, file_descr_id, created_at, updated_at) VALUES (?,?,?,?,?,?,?,?)"
-    (fileId, FSNew, fileConnReq, fileInline, rcvFileInline, rfdId, currentTs, currentTs)
-  pure RcvFileTransfer {fileId, fileInvitation = f, fileStatus = RFSNew, rcvFileInline, rcvFileDescription = rfd, senderDisplayName = c, chunkSize, cancelled = False, grpMemberId = Nothing}
+  case fileDescr of
```
Comment (epoberezkin) [inline src/Simplex/Chat/Store.hs:2909]: it seems like it's better to split it to two functions if it is not possible to somehow refactor it...

File: src/Simplex/Chat/Store.hs [hunk 5]
```diff
@@ -2896,47 +2898,122 @@ deleteSndFileChunks :: DB.Connection -> SndFileTransfer -> IO ()
...
+      liftIO $
+        DB.execute
+          db
+          "INSERT INTO rcv_files (file_id, file_status, file_queue_info, file_inline, rcv_file_inline, file_descr_id, created_at, updated_at) VALUES (?,?,?,?,?,?,?,?)"
+          (fileId, FSNew, fileConnReq, fileInline, rcvFileInline, rfdId, currentTs, currentTs)
+      pure RcvFileTransfer {fileId, fileInvitation = f, fileStatus = RFSNew, rcvFileInline, rcvFileDescription = Just rfd, senderDisplayName = c, chunkSize, cancelled = False, grpMemberId = Nothing}
+    Nothing -> do
+      DB.execute
+        db
+        "INSERT INTO rcv_files (file_id, file_status, file_queue_info, file_inline, rcv_file_inline, created_at, updated_at) VALUES (?,?,?,?,?,?,?)"
```
Comment (epoberezkin) [inline src/Simplex/Chat/Store.hs:2922]: although this query is exactly the same, other than description ID that can be passed - not sure why it needs to be repeated.

File: src/Simplex/Chat/Store.hs [hunk 6]
```diff
@@ -2896,47 +2898,122 @@ deleteSndFileChunks :: DB.Connection -> SndFileTransfer -> IO ()
...
+      liftIO $
+        DB.execute
+          db
+          "INSERT INTO rcv_files (file_id, file_status, file_queue_info, file_inline, rcv_file_inline, group_member_id, file_descr_id, created_at, updated_at) VALUES (?,?,?,?,?,?,?,?,?)"
+          (fileId, FSNew, fileConnReq, fileInline, rcvFileInline, groupMemberId, rfdId, currentTs, currentTs)
+      pure RcvFileTransfer {fileId, fileInvitation = f, fileStatus = RFSNew, rcvFileInline, rcvFileDescription = Just rfd, senderDisplayName = c, chunkSize, cancelled = False, grpMemberId = Just groupMemberId}
+    Nothing -> do
+      DB.execute
+        db
+        "INSERT INTO rcv_files (file_id, file_status, file_queue_info, file_inline, rcv_file_inline, group_member_id, created_at, updated_at) VALUES (?,?,?,?,?,?,?,?)"
```
Comment (epoberezkin) [inline src/Simplex/Chat/Store.hs:2947]: same here

File: src/Simplex/Chat/Store.hs (hunk 1 above)
Comment (epoberezkin) [reply src/Simplex/Chat/Store.hs:3003]: yes

**Review by epoberezkin — APPROVED** [review]

**Pull request merged**